src/monodroid/jni/android-system.cc

```diff
@@ -616,22 +627,10 @@ AndroidSystem::get_existing_dso_path_on_disk (const char *base_dir, const char *
 		return dso_path;
 
 	*needs_free = false;
-	free (dso_path);
+	delete[] dso_path;
 	return nullptr;
 }
 
-void
-AndroidSystem::dso_alloc_cleanup (char **dso_path, bool *needs_free)
```

Comment:
How/why is `dso_alloc_cleanup()` no longer needed?

Reply:
If you look at the old code, it was never really needed :) (I think it remained from some earlier rewrite, as an oversight):

```c++
if (dso_path != nullptr)
      return dso_path;
dso_alloc_cleanup (&dso_path, needs_free);
```

There wasn't ever anything to clean up. We do delete the pointer at the call site, as it should be, now.